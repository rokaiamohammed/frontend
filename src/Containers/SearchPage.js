import React, { Component }  from 'react';
// import logo from './logo.svg';
import './SearchPage.css';
import CardList from '../Components/SearchCardlist'
import {types} from '../Components/Searchtypes'
import Card from '../Components/SearchCard'

class SearchPage extends Component{
  constructor(){
    super();
    this.state={
      types:[],
      clickid:"7",
      topcardname:'',
      id:''
    }

  }

  componentWillMount(){
    var valueArray = [];
    const requestOptions = {
      method: 'GET',
      headers: { 
      'Content-Type': 'application/x-www-form-urlencoded',  
      'authorization': sessionStorage.getItem('token'),
      'Accept': 'application/json'},
      }
    fetch('http://13.72.75.179/api/browse ',requestOptions)
    .then(response=>response.json())
    .then(function(res) {
      res.forEach(element => {
          valueArray.push(element);
});
})

.then( this.setState({types:valueArray}))
.then(()=>this.setState({topcardname:this.state.types[7].name}))
.then(()=>this.setState({id:"7"}))
 
  }

  render(){
    return(
      <div className="vh-100 dt w-100 pl2">
        <h1 className="pl4 fw7 pt4 f3 lh-title white  ">Your top genres</h1>
         <div >
          <Card idc={this.state.id}
            title={this.topcardname}
            bg={this.topcardname}
            />
           </div>
        <h1 className=" pl4 fw7 pt4 f3 lh-title white ">Browse all</h1>
        <CardList list={this.state.types} EVENT={this.ondivclick}/>
        
    </div>

    )
  }

}

export default SearchPage;
