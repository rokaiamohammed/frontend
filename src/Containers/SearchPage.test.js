// import React from 'react';
// import { Provider } from 'react-redux';
// import { shallow } from 'enzyme';
import SearchPage from './SearchPage';

// import { act } from "react-dom/test-utils";

// import { render, unmountComponentAtNode } from "react-dom";

// let container;
// beforeEach(() => {
//   // setup a DOM element as a render target
//   container = document.createElement("div");
//   document.body.appendChild(container);
// });

// afterEach(() => {
//   // cleanup on exiting
//   unmountComponentAtNode(container);
//   container.remove();
//   container = null;
// });


// it('sets isLoading to true when loading', () => {
//   global.fetch = jest.fn().mockReturnValue(Promise.resolve({
//     ok: true,
//     json: () => Promise.resolve({'user': 'test'})
//   }));
//  expect(global.fetch).toBeCalledWith('http://localhost:5000/browse')
//   const c = shallow(<SearchPage />);
//   c.find("Card").prop('title')();
//   expect(c.instance().state.loaded).toEqual(true);
//   // await flushPromises();
//   expect(c.instance().state.isLoading).toEqual(false);
// });
// it('calls api for correct username', () => {
//   global.fetch = jest.fn().mockReturnValue(Promise.resolve({
//     ok: true,
//     json: () => Promise.resolve({'user': 'test'})
//   }));

//   // c.find('CardList').prop('list')();
 
// });



// // describe("User component", () => {
// // test("renders user data", async () => {
// //   const fake=[{ name: "John Doe" }, { name: "Kevin Mitnick" }];
// //   jest.spyOn(window, "fetch").mockImplementation(() => {
// //     const fetchResponse = {
// //       json: () => Promise.resolve(fake)
// //     };
// //     return Promise.resolve(fetchResponse);
// //   });
// //   // Use the asynchronous version of act to apply resolved promises
// //   await act(async () => {
// //     render(<SearchPage />, container);
// //   });
// //   const wrapper =shallow(<SearchPage/>)
// //   const text = wrapper.find('Card').title()
// //   expect(text).toEqual('Current count: 0')
// //   // expect(container.).toBe("John DoeKevin Mitnick")
// //   window.fetch.mockRestore();
// //   // remove the mock to ensure tests are completely isolated 
// // });
// // });


import React from 'react';
import CardList from '../Components/SearchCardlist';

import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Card from '../Components/SearchCard';

configure({adapter: new Adapter()});

describe("<searchPage />", () => {
  let wrapper;

  beforeEach(() => {
      wrapper = shallow(<SearchPage />);
  });

  it('searchPage', () => {
    wrapper.setState({
      types:[{name: "John Doe"}],
      clickid:"7",
      topcardname:'Pop',
      id:'2',
    }); 
    wrapper=wrapper.update();
    expect(wrapper.containsAnyMatchingElements([
    <h1>Your top genres</h1>,<Card title="Pop" idc="2"/>,<h1>Browse all</h1>,<CardList list="[{name: 'John Doe'}]"/>])).toEqual(true);
 
  });
});