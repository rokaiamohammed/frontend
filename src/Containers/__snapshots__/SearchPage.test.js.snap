// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fileters Robots correctly 1`] = `ShallowWrapper {}`;

exports[`renders with or without a name 1`] = `ShallowWrapper {}`;

exports[`renders without crashing 1`] = `ShallowWrapper {}`;
